tests/databases/utilities.py

```diff
@@ -0,0 +1,130 @@
+def assert_database_state(state1, state2):
```

Comment:
I'm confused what this is adding? Is it just so it's clearer what you're doing when you compare two databases?

Reply:
Pretty much, it also means that in cases where errors happen you get additional info, because state1 and state2 get printed.